add debugging rails apps documentation (#2)
hmdc/ondemand_dev_guide: RAILS_PASSENGER_APPS_GUIDE.md

which means that it will appear under the `Files` dropdown menu with the name `Rails Prototype` and the
specified icon.

Further documentation on **Manifest** files can be found [here](https://osc.github.io/ood-documentation/latest/how-tos/app-development/interactive/manifest.html?highlight=manifest)

# Developing in your local environment

> You can develop the Rails application as a standalone normal Rails app with your IDE. 

Reviewer: ```suggestion
You can develop the application as a standalone Rails app within your IDE. 
```

> - You can run the application in development environment by running this in the console:

Reviewer: ```suggestion
- To run the application in development mode execute the following command in the console:
```

```bash
rails server
```

The application will be served in `localhost:3000`

Alternatively, if you use RubyMine, you can click on the `Play` button in the top-right bar.

## Debugging Rails applications

### Debugging with RubyMine

> If you use RubyMine IDE, it contains its own debugging capabilities. To do that, you can place a breakpoint in

Reviewer: ```suggestion
RubyMine contains its own debugging capabilities. To start debugging, place a breakpoint in
```

the line of code you want to debug:

![breakpoint](images/rails_passenger_apps_guide/02_debugger_rubymine_breakpoints.png)

And then click on the `Debug` button on the top-right bar.

> Once you make a request that runs that piece of code, the debugger will stop the execution by showing the 

Reviewer: ```suggestion
Make a request with your browser that runs that line of code, the debugger will stop the execution by showing the 
```

status of the process.

![debugger working](images/rails_passenger_apps_guide/03_debugger_rubymine_stopped.png)

### IDE-independent debugging

Rails has also IDE-independent debugging capabilities with `debug` and `web-console` gems, included in the 
default Rails 7.2 applications.

> - To use `debug`, you need to add the `debug` gem in your Gemfile and then place the `debugger` method in the piece of 

Reviewer: ```suggestion
- To use `debug`, add the `debug` gem in your Gemfile and then place the `debugger` statement in the within the 
```

code you want to debug and execute the Rails application. Rails Server will stop the execution in that point and you 
will be able to debug from the terminal, by even invoking `irb` to inspect the variables:

![debug gem](images/rails_passenger_apps_guide/04_debugger_debug_gem.png)

> - To use `web-console`, you need to declare that gem in the Gemfile and put `console` in the piece

Reviewer: ```suggestion
- To use `web-console`, you need to declare that gem in the Gemfile and add the `console` statement within the
```

> of code you want to debug. A Rails Console will be open in the web browser, that will allow you to

Reviewer: ```suggestion
code you want to debug. A Rails Console will be open in the web browser, that will allow you to
```
